### PR TITLE
Add administrative and technical contacts to replace contact info

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This pull request adds `administrative_contact` and `technical_contact` to applications, OAuth clients, gateways and organizations. ~It also adds partial Console support.~

Refs #4930.

#### Changes

- Define API
- Define Database Models in the IS
- Modify stores to handle creates/updates with contacts
- Modify Update RPCs to handle creates/updates with contacts
- Update CLI to allow un-setting contacts

#### Testing

- Updated IS unit tests

##### Regressions

n/a

#### Notes for Reviewers

nil

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by kschiffer** [comment]
Let's move the settings for admin and tech contact to the application/gateway general settings to keep creation forms lean. It is sufficient to be able to set this setting after creation. By default, I think we can assume that the initial owner is admin and tech contact or does something speak against that?

**Comment by kschiffer** [comment]
I've created an issue for a better way of selecting users/orgs here: #5027

**Comment by htdvisser** [comment]
I'm going to rebase and force-push this branch so that it doesn't diverge too much.

**Comment by htdvisser** [comment]
In order to move this forward, I've removed my commits that implemented _setting_ entity contacts in the Console. Hopefully somewhere in the future our frontend devs have time to implement that properly, and otherwise I can try to spend some more time on it after #4935 and #4936.

**Review by johanstokking — APPROVED** [review]
API changes LGTM

**Comment by htdvisser** [comment]
Added two additional commits to:

1. Always set the collaborator as entity contact on create
2. Set the first organization collaborator, and otherwise the first user collaborator, as entity contact for existing entities (in a migration)

For convenience, here's what the migration for applications looks like:

```sql
WITH application_owners AS (
    SELECT
        applications.id application_uuid,
        accounts.id account_uuid,
        FIRST_VALUE (accounts.id) OVER (
            PARTITION BY applications.id
            ORDER BY
                accounts.account_type ASC, -- Prefer organization contacts over user contacts
                memberships.created_at     -- Prefer the oldest collaborator
        ) first_account_uuid
    FROM
        accounts
        JOIN memberships ON accounts.id = memberships.account_id
        JOIN applications ON applications.id = memberships.entity_id
        AND memberships.entity_type = 'application'
    WHERE
        accounts.deleted_at IS NULL
        AND memberships.rights && ARRAY [
    28, -- RIGHT_{ENTITY}_ALL
    55  -- RIGHT_ALL
  ]
)
UPDATE
    applications
SET
    administrative_contact_id = application_owners.account_uuid,
    technical_contact_id = application_owners.account_uuid
FROM
    application_owners
WHERE
    applications.id = application_owners.application_uuid
    AND applications.administrative_contact_id IS NULL
    AND applications.technical_contact_id IS NULL
    AND application_owners.account_uuid = application_owners.first_account_uuid;
```

File: pkg/webui/console/components/contact/index.js
Comment (bafonins) [inline pkg/webui/console/components/contact/index.js]:
```suggestion
        <Message className="tc-subtle-gray" component="i" content={sharedMessages.none} />
```

File: pkg/webui/console/components/contact/index.js
Comment (bafonins) [inline pkg/webui/console/components/contact/index.js]:
```suggestion
  const id = getCollaboratorId({ids: props.ids})
```

from https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.17/pkg/webui/lib/selectors/id.js#L46-L48

File: pkg/webui/console/components/contact/index.js
Comment (bafonins) [inline pkg/webui/console/components/contact/index.js]:
```suggestion
      <Icon icon={icon} className={style.collaboratorIcon} nudgeUp /> {id}
```

**Review by bafonins — CHANGES_REQUESTED** [review]

**Review by kschiffer — COMMENTED** [review]
While the code looks good to me (with @bafonins remarks), I think that the utility of the new contact info is quite limited:
- Only the names of the orgs/users is provided without click-through
- We miss public profiles of orgs and users, which makes it impossible for non-admin users to watch/interact with other users
- Organizations often have no contact info associated with them, making it impossible to contact

We should at least add the public contact info to the name as a link (if present) next to the name.

File: pkg/webui/console/components/contact/index.js
Comment (bafonins) [inline pkg/webui/console/components/contact/index.js]: You can remove the wrapping `<i />`. `component="i"` on L31 `<Message />` will be rendered as `<i />`

**Review by bafonins — APPROVED** [review]
See my comment. `webui` changes look good, but please wait for @kschiffer approval as well

**Comment by htdvisser** [comment]
@kschiffer already gave an LGTM on the code yesterday, but at this point I can't address his other suggestions. So if those are blockers for his approval, I can just omit the Console commits entirely.

This already has taken more time than it should have, and I want to get this merged today.

**Review by kschiffer — CHANGES_REQUESTED** [review]
Let's please extract the console part from this then for now and move it to a branch for reference as I am afraid that in its current form it creates more problems than it solves.

**Review by kschiffer — APPROVED** [review]

**Pull request merged**
